packages/strapi-design-system/src/Select/SingleSelect.tsx

```diff
@@ -0,0 +1,330 @@
+  children,
+  clearLabel = 'Clear',
+  customizeContent = (value) => value.toString(),
+  disabled,
+  error,
+  hint,
+  id,
+  label,
+  labelAction,
+  onChange,
+  onClear,
+  onReachEnd,
+  placeholder,
+  required,
+  selectButtonTitle,
+  startIcon,
+  size = 'M',
+  value: passedValue,
+}: SingleSelectProps) => {
+  /**
+   * Used for the intersection observer
+   */
+  const viewportRef = React.useRef<HTMLDivElement>(null);
+
+  const clearRef = React.useRef(null);
+
+  /**
+   * These values are drawn out from the internals of the Radix component
+   * We can then use them to react to visual changes for the component
+   */
+  const [internalValue, setInternalValue] = React.useState('');
+  const [internalIsOpen, setInternalIsOpen] = React.useState(false);
+
+  const generatedId = useId('select', id);
+
+  const hintId = `${generatedId}-hint`;
+  const errorId = `${generatedId}-error`;
+
+  const handleValueChange: Pick<RadixSelect.SelectProps, 'onValueChange'>['onValueChange'] = (value) => {
+    /**
+     * If it's being externally managed then we shouldn't
+     * both setting our copy of the internal value.
+     */
+    if (onChange) {
+      const shouldBeNumber = typeof passedValue === 'number';
+      onChange(shouldBeNumber ? Number(value) : value);
+    } else {
+      setInternalValue(value);
+    }
+  };
+
+  const handleClearClick: React.MouseEventHandler<HTMLDivElement> = (e) => {
+    console.log('calling clear');
+    if (onClear) {
+      onClear(e);
+    }
+  };
+
+  const handleOpenChange: Pick<RadixSelect.SelectProps, 'onOpenChange'>['onOpenChange'] = (open) => {
+    setInternalIsOpen(open);
+  };
+
+  /**
+   * We need to do this so the clear button can actually be
+   * used as opposed to the trigger swallowing the event.
+   *
+   * I'm not proud of it, but people are asking for it so maybe
+   * we can remove this in the future – https://github.com/radix-ui/primitives/issues/1569
+   */
+  const handleTriggerPointerDown: React.PointerEventHandler<HTMLButtonElement> = (e) => {
+    // @ts-ignore
+    if (clearRef.current && clearRef.current === e.target.closest('div')) {
+      console.log('preventing default');
+      e.preventDefault();
+    }
+  };
+
+  const intersectionId = `intersection-${generatedId}`;
+
+  const handleReachEnd = (entry: IntersectionObserverEntry) => {
+    if (onReachEnd) {
+      onReachEnd(entry);
+    }
+  };
+
+  useIntersection(viewportRef, handleReachEnd, {
+    selectorToWatch: `#${intersectionId}`,
+    /**
+     * We need to know when the select is open because only then will viewportRef
+     * not be null. Because it uses a portal that (sensibly) is not mounted 24/7.
+     */
+    skipWhen: !internalIsOpen,
+  });
+
+  const value = typeof passedValue !== 'undefined' ? passedValue.toString() : internalValue;
+
+  return (
+    <Field hint={hint} error={error} id={generatedId} required={required}>
+      <Stack spacing={label || hint || typeof error === 'string' ? 1 : 0}>
```

Comment:
`label` is a required prop so this will always be truthy.

Reply:
Ah no ... it is not?! 🤔

Reply:
Actually i have made it required now... so yeah it'll always be truthy.